atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/AsyncCellLoader.java

```diff
@@ -73,29 +76,39 @@ private AsyncCellLoader(CqlClient cqlClient, Executor executor, String keyspace)
                                 .filter(Optional::isPresent)
                                 .map(Optional::get)
                                 .collectToMap(),
-                        executor);
+                        executorService);
     }
 
     private ListenableFuture<Optional<Value>> loadCellWithTimestamp(
             TableReference tableRef,
             Cell cell,
             long timestamp) {
+        CqlQueryContext queryContext = ImmutableCqlQueryContext.builder()
+                .tableReference(tableRef)
+                .keyspace(keyspace)
+                .build();
+        GetQueryParameters getQueryParameters = ImmutableGetQueryParameters.builder()
+                .cell(cell)
+                .humanReadableTimestamp(timestamp)
+                .build();
+
         return cqlClient.executeQuery(
                 ImmutableGetQuerySpec.builder()
-                        .tableReference(tableRef)
-                        .keySpace(keyspace)
-                        .row(ByteBuffer.wrap(cell.getRowName()))
-                        .column(ByteBuffer.wrap(cell.getColumnName()))
-                        .humanReadableTimestamp(timestamp)
-                        .executor(executor)
+                        .cqlQueryContext(queryContext)
+                        .queryParameters(getQueryParameters)
                         .build());
     }
 
-    private static  <V> V getDone(Future<V> listenableFuture) {
+    private static <V> V getDone(Future<V> listenableFuture) {
         try {
             return Futures.getDone(listenableFuture);
         } catch (ExecutionException e) {
             throw new RuntimeException(e.getCause());
         }
     }
+
+    @Override
+    public void close() {
+        executorService.shutdown();
```

Comment:
Something feels weird here, why is it closing the ExecutorService and not the client?

Reply:
Closing is now more centralized.